### PR TITLE
fixes #13828 - CVE-2016-2100 - only showing relevant bookmarks

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

**Comment by domcleal** [comment]
This only changes the `index` action, all of the others remain vulnerable, as does the API controller.  Use resource_base etc, and please check the ticket comments for all of the other unfixed issues here.

The PR is also missing tests.

**Comment by domcleal** [comment]
Thanks, looks better except that the API controllers look unfixed.  Could you do the same there?

File: app/controllers/concerns/bookmark_common.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module BookmarkCommon
+  def resource_base
+    super.my_bookmarks
+  end
+
+  def resource_scope(*args)
+    super.my_bookmarks
+  end
+end
```
Comment (shlomizadok) [inline app/controllers/concerns/bookmark_common.rb]: Nit: EOL

**Comment by shlomizadok** [comment]
:+1:

File: app/controllers/concerns/bookmark_common.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module BookmarkCommon
```
Comment (domcleal) [inline app/controllers/concerns/bookmark_common.rb]: Usually these are in the Foreman::Controller namespace, see other files in the same dir.

**Comment by domcleal** [comment]
There should be tests for the API changes too.

**Comment by domcleal** [comment]
Merged as a61344d, thanks @unorthodoxgeek.

**Pull request closed without merging**